doc/development.md

```diff
@@ -0,0 +1,120 @@
+# Crowbar: Development
+
+This document describes the recommended setup for Crowbar front-end development
+(eg. CSS, SASS, HAML, HTML, Rails).
+
+## Background
+
+For designers and (especially front-end) developers to have a productive
+workflow, it is critical to be able to work directly from a Git checkout and
+have, as much as possible, the exact same UI/UX as the actual product (and
+upstream).
+
+The challenge is that Crowbar is a complex product with many moving parts and
+dependencies. The code is not structured or designed to run out of a Git
+checkout. Many improvements have been made to simplify the running of the
+Crowbar web UI from a Git checkout, but for a fully functional system (which one
+needs to see and test all the UI and interactions), there is still a quite some
+work.
+
+In the following section we will describe in detail, the requirements of this
+development setup and how it is configured.
+
+## Flavors
+
+We have the following two flavors of the Crowbar Web UI running so we can easily
+compare the differences:
+
+  1. __Regular SUSE Cloud install__. Runs on [port 3000]
+     (http://HOSTNAME:3000) and from the `/opt/dell` directory.
+  2. __From Git upstream branding__. Runs on [port 5000]
+     (http://HOSTNAME:5100) and from the `/opt/crowbar` directory.
+
+All are using the same default username and password, both ```crowbar```. The
+following sections will describe in detail how each of these are setup and
+configured.
+
+### Installing
+
+Before we can start you need to match some prerequirements on your host machine.
+
+  1. Please create this script somewhere in your path, it's just a wrapper
+     around our mkcloud script that will be mentioned later on again. We place
+     it in general at /usr/local/bin/mkcloud6
+
+     ```bash
+     #!/usr/bin/env bash
+
+     export pre_onadmin_prepareinstallcrowbar=$(base64 --wrap=0 <<'EOS'
+       zypper ar -f http://download.suse.de/ibs/Devel:/Cloud:/Shared:/Rubygem/SLE_11_SP3/Devel:Cloud:Shared:Rubygem.repo;
+       zypper mr -r -e sp3sdk;
```

Comment:
not sure why but this `mr` hasn't been done when i set up my cloud with the hook... therefore the `in` wasn't successful either. i had to do it manually again. maybe a `zypper ref` is still necessary here.